test/test_shared_library.py

```diff
@@ -0,0 +1,98 @@
+from translator.source_code import translator_shared_library
+import json
+import pytest
+import unittest
+
+#--------------------------------------------------------------------------------unit test for valid_info function--------------------------------------------------------------------------------
+def test_valid_info_valid_info():
+  test_info = {
+    'feed_info_id': "104d7746-688c-44ed-b195-2ee948bf9dfa",
+    'metadata':{
+      'wz_location_method': "channel-device-method",
+      'lrs_type': "lrs_type",
+      'contact_name':"Abinash Konersman",
+      'contact_email': "abinash.konersman@state.co.us",
+      'issuing_organization': "iCone",
+      }
+    }
+  test_validate_info=translator_shared_library.validate_info(test_info)
+  assert test_validate_info == True
+
+def test_valid_info_no_info():
+  test_info = None
+  test_validate_info=translator_shared_library.validate_info(test_info)
+  assert test_validate_info == False
+
+def test_valid_info_invalid_info_missing_required_fields_lrs_type():
+  test_info = {
+    'feed_info_id': "104d7746-688c-44ed-b195-2ee948bf9dfa",
+    'metadata':{
+      'wz_location_method': "channel-device-method",
+      'contact_name':"Abinash Konersman",
+      'contact_email': "abinash.konersman@state.co.us",
+      'issuing_organization': "iCone",
+      }
+    }
+  test_validate_info=translator_shared_library.validate_info(test_info)
+  assert test_validate_info == False
+
+def test_valid_info_invalid_info_invalid_feed_info_id(): 
+  test_info = {
+    'feed_info_id': "104d7746-e948bf9dfa",
+    'metadata':{
+      'wz_location_method': "channel-device-method",
+      'lrs_type': "lrs_type",
+      'contact_name':"Abinash Konersman",
+      'contact_email': "abinash.konersman@state.co.us",
+      'issuing_organization': "iCone",
+      }
+    } 
+  test_validate_info=translator_shared_library.validate_info(test_info)
+  assert test_validate_info == False
+
+#--------------------------------------------------------------------------------unit test for parse_xml function--------------------------------------------------------------------------------
+def test_parse_xml():
```

Comment:
Good success case, can we add a check for failed cases?

Reply:
added some unit test

Reply:
Looks like this added a case where the passed in value is empty. Can we add one more to test with a non-existent file name since we're checking for that in the code? IE something like 'non-file.txt'.

Reply:
added another test case for non-file.txt